another approach to testing interceptor redirects

jeffbrown opened the pull request:
This may serve as a starting point and example for how to try and express whatever redirect bug you have found but cannot demonstrate.  This is setup just as a demo app and is doing things that I would not do in a real app, but does facilitate tracking what the interceptor is doing.  The tracking is done by managing state in the session which acts kind of as a bread trail for where control has been in the interceptor.  You should be able to adapt this with very minor changes to demonstrate the bug that you have found.  Once you have done that, please let us know.

orubel commented:
Was there a reason you tied the interceptor directly to a controller?

+ match uri: "/$entrypoint/post/**"

The interceptor was previously intercepting all controller/action traffic
as long as it matching that uri. Now you have it matching a single
controller



Owen Rubel
415-971-0976
orubel@gmail.com

On Sat, Aug 22, 2015 at 5:40 AM, Jeff Scott Brown <notifications@github.com>
wrote:

> This may serve as a starting point and example for how to try and express
> whatever redirect bug you have found but cannot demonstrate. This is setup
> just as a demo app and is doing things that I would not do in a real app,
> but does facilitate tracking what the interceptor is doing. The tracking is
> done by managing state in the session which acts kind of as a bread trail
> for where control has been in the interceptor. You should be able to adapt
> this with very minor changes to demonstrate the bug that you have found.
> Once you have done that, please let us know.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/orubel/testapp/pull/10
> Commit Summary
>
>    - another approach to testing interceptor redirects
>
> File Changes
>
>    - *A* grails-app/controllers/MessageController.groovy
>    <https://github.com/orubel/testapp/pull/10/files#diff-0> (7)
>    - *M* grails-app/controllers/TestappInterceptor.groovy
>    <https://github.com/orubel/testapp/pull/10/files#diff-1> (52)
>    - *M* grails-app/controllers/UrlMappings.groovy
>    <https://github.com/orubel/testapp/pull/10/files#diff-2> (6)
>    - *M*
>    src/integration-test/groovy/testapp/MappingConfigFunctionalSpec.groovy
>    <https://github.com/orubel/testapp/pull/10/files#diff-3> (195)
>
> Patch Links:
>
>    - https://github.com/orubel/testapp/pull/10.patch
>    - https://github.com/orubel/testapp/pull/10.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/orubel/testapp/pull/10>.
>

jeffbrown commented:
> Was there a reason you tied the interceptor directly to a controller?

The reason that I changed the uri pattern was because it was a very simple way to not have the interceptor apply to the `MessageController` that I introduced.  I don't want the interceptor engaged in requests to that controller.  That controller is only there to support the simple functional test I wrote.  I wouldn't do anything like that in a real app but it is a simple practical way to provide a functional test that can easily track what is going on in the interceptor.

orubel commented:
You are totally 100% correct. In a production env you would never want the requests for message controller to route through here; thats just silly because its additional overhead.

I'm trying to assume all controllers would route through this controller (rather than having to hand code one per). So I changed it so that I did a detect on params.controller internally for 'message' to avoid that functionality being applied when message controller is applied.

But you are 100% correct. I just am trying to simulate an environment where I have (X) controllers.

But again, you are definitely correct for what you are doing.
